### PR TITLE
check % disabled ports before per-port-polling

### PR DESCRIPTION
**Pull request by barryodonovan**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

File: doc/Support/Performance.md [hunk 1]
```diff
@@ -62,30 +62,32 @@ You can also set this globally with the config option `$config['snmp']['max_oid'
...
-The default 16 threads that `poller-wrapper.py` runs as isn't necessarily the optimal number. A general rule of thumb is 
+The default 16 threads that `poller-wrapper.py` runs as isn't necessarily the optimal number. A general rule of thumb is
 2 threads per core but we suggest that you play around with lowering / increasing the number until you get the optimal value.
 
 This can be changed by going to the cron job for librenms. Usually in /etc/cron.d/librenms and changing the "16"
 
 */5  *    * * *   librenms    /opt/librenms/cronic /opt/librenms/poller-wrapper.py 16
 
-KEEP in MIND that this dosnt always help, it depnds on your system and CPU. So Be careful. 
+KEEP in MIND that this dosnt always help, it depnds on your system and CPU. So Be careful.
```
Comment (murrant) [inline doc/Support/Performance.md]: dosnt -> does not

File: includes/polling/ports.inc.php [hunk 2]
```diff
@@ -176,7 +176,7 @@
 if ($device['os'] === 'f5' && (version_compare($device['version'], '11.2.0', '>=') && version_compare($device['version'], '11.7', '<'))) {
     require_once 'ports/f5.inc.php';
 } else {
-    if ($config['polling']['selected_ports'] === true || $device['attribs']['selected_ports'] == 'true') {
+    if ($config['polling']['selected_ports'] === true || $device['attribs']['selected_ports'] == 'true' && dbFetchRow("SELECT (count(*) / (SELECT count(*) from `ports` WHERE `device_id` = ?))> 0.33 FROM `ports` WHERE `device_id`= ? AND `deleted` = 1 HAVING count(*) > 10", array($device['device_id'],$device['device_id']))) {
```
Comment (murrant) [inline includes/polling/ports.inc.php]: Use dbFetchCell() to get a single result.

**Comment by murrant** [comment]
I think you should take ifAdminStatus and disabled into account too.

Perhaps extract the variables to settings so they are tunable.

I think we need to collect some data.

I came up with this query to show port stats, but I think we need the time it takes on each device for the full poll and per-port poll:

```
SELECT
  `op`.`device_id`,
  (SELECT `os` FROM `devices` WHERE `device_id` = `op`.`device_id`) AS `os`,
  count(*) AS `total`,
  ((SELECT count(*) from `ports`
     WHERE `device_id` = `op`.`device_id` AND
     (`deleted` = 1 OR `ifAdminStatus` != 'up' OR `disabled` = 1)
   ) / count(*)) AS `ratio`
FROM `ports` AS `op`
GROUP BY `op`.`device_id` ORDER BY `ratio` DESC;
```

File: doc/Support/Performance.md (hunk 1 above)
Comment (barryodonovan) [reply doc/Support/Performance.md]: Also a typo in `depnds` but I don't know how to pull your changes back into mine...

**Comment by barryodonovan** [comment]
image
@murrant 
This is one device, a mikrotik router with ~60 pppoe sessions and 13 ports. You can see the point at which I turned on per-port

**Comment by murrant** [comment]
I made script to collect stats, useful?  #7626

File: includes/polling/ports.inc.php [hunk 3]
```diff
@@ -176,7 +176,7 @@
 if ($device['os'] === 'f5' && (version_compare($device['version'], '11.2.0', '>=') && version_compare($device['version'], '11.7', '<'))) {
     require_once 'ports/f5.inc.php';
 } else {
-    if ($config['polling']['selected_ports'] === true || $device['attribs']['selected_ports'] == 'true') {
+    if ($config['polling']['selected_ports'] === true || $device['attribs']['selected_ports'] == 'true' && dbFetchCell("SELECT (count(*) / (SELECT count(*) from `ports` WHERE `device_id` = ?))> 0.33 FROM `ports` WHERE `device_id`= ? AND `deleted` = 1 HAVING count(*) > 10", array($device['device_id'],$device['device_id']))) {
```
Comment (Zmegolaz) [inline includes/polling/ports.inc.php]: Could you move the SQL outside of the if(), to make it easier to read?

**Comment by laf** [comment]
I'm proposing that we close this.

 Overall I think it's definitely the right approach that we try and work out the best options automatically but I think we need to do it in a structure way so we can re-use the code for more than just selected port polling. @librenms/reviewers

**Comment by murrant** [comment]
Sounds good to me.  Let's revisit this at a later time.

**Pull request closed without merging**